swri_roscpp/include/swri_roscpp/dynamic_parameters.h

```diff
@@ -392,6 +426,34 @@ namespace swri
         desc.level = 0;
         desc.description = param->second.description;
         desc.edit_method = "";
+
+        // If this is an enum, lets make the edit method string
+        for (int i = 0; i < param->second.enums.size(); i++)
```

Comment:
This looks like it is shadowing the `i` parameter from the outer loop, is that intentional?  And is there a reason to not use for-each loops here?

Reply:
Ooops. Should not do that.

I wanted an index so writing out the loop manually is usually the most concise way of doing that.

Reply:
Also, it seems some people are requesting this to support C++03 which doesnt allow those.